### PR TITLE
fix: Avoid multiple event handlers of the same instance

### PR DESCRIPTION
**Pull request by barak007**
Probably breaking change

### CODE REVIEW

File: packages/core/test/node/remote-service-com.spec.ts
Comment (RomanYarik) [inline packages/core/test/node/remote-service-com.spec.ts:89]: why not use chai-as-promised?
expect(proxy.on(testListenerStub)).to.eventually.rejectWith('..')

**Review by RomanYarik — APPROVED** [review]

File: packages/core/test/node/remote-service-com.spec.ts
Comment (barak007) [reply packages/core/test/node/remote-service-com.spec.ts]: I forgot we have this. changed

**Review by RomanYarik — APPROVED** [review]

**Comment by barak007** [comment]
@RomanYarik after running the test in `wcs` It seems that this will not break it. So we can probably merge it as patch.
There is still the possibility of this breaking during work but I'm not sure how to test it. 
@Ari-Kishon can we make sure we play with `wcs` after this merge and try to see if there is any new errors? (if there are we want to fix them quickly)

**Comment by Ari-Kishon** [comment]
@PeterShershov 
should we validate and merge this before or after the release?

**Comment by barak007** [comment]
@Ari-Kishon 
I did a lot more testing and everything looks good. I think it's much safer to release. 
One thing. I did all my tests on the web version and not electron but I don't expect it to be deferent.

**Pull request merged**
